Refresh.GameServer/Endpoints/Game/Playlists/PlaylistEndpoints.cs

```diff
@@ -0,0 +1,338 @@
+    {
+        // Extract the slot type, ensuring its set
+        string? slotType = context.QueryString["slot_type"];
+        if (slotType == null)
+            return BadRequest;
+        
+        // Extract the slot ID ensuring its valid
+        if (!int.TryParse(context.QueryString["slot_id"], out int slotId))
+            return BadRequest;
+
+        GamePlaylist? parentPlaylist = database.GetPlaylistById(playlistId);
+
+        // If the parent doesn't exist, exit gracefully
+        if (parentPlaylist == null)
+            return NotFound;
+
+        // Dont let people add slots to other's playlists
+        if (parentPlaylist.Creator.UserId != user.UserId)
+            return Unauthorized;
+
+        // Special handling for playlists
+        if (slotType == "playlist")
+        {
+            GamePlaylist? childPlaylist = database.GetPlaylistById(slotId);
+
+            // If the child doesn't exist, exit gracefully
+            if (childPlaylist == null)
+                return NotFound;
+
+            // Dont allow a playlist to be a child of itself
+            if (childPlaylist.PlaylistId == parentPlaylist.PlaylistId)
+                return BadRequest;
+
+            // If the parent contains the child in its parent tree, block the request to prevent recursive playlists
+            // This would be a `BadRequest`, but the game has a bug and will do this when creating sub-playlists,
+            // so lets not upset it and just return OK, I dont expect this to be a common problem for people to run into.
+            bool recursive = false;
+            parentPlaylist.TraverseParentsRecursively(database, delegate(GamePlaylist playlist)
+            {
+                if (playlist.PlaylistId == childPlaylist.PlaylistId)
+                    recursive = true;   
+            });
+            if (recursive) return OK;
+            
+            // Add the playlist to the parent   
+            database.AddPlaylistToPlaylist(childPlaylist, parentPlaylist);
+
+            // ReSharper disable once ExtractCommonBranchingCode see like 3 lines below (line count subject to change)
+            return OK;
+        }
+        // ReSharper disable once RedundantIfElseBlock i am intentionally writing this code like this to prevent code
+        //                                             accidentally falling outside of the branch during a possible future
+        //                                             refactor and returning OK when not intended. ok, rider? 
+        else
+        {
+            GameLevel? level = database.GetLevelByIdAndType(slotType, slotId);
+
+            // If the level doesn't exist, exit gracefully
+            if (level == null)
+                return NotFound;
+
+            // Add the level to the playlist    
+            database.AddLevelToPlaylist(level, parentPlaylist);
+            
+            return OK;
```

Comment:
If you find yourself using the exact same wording, but with English grammar instead of 'C# grammar', e.g. commenting 'Add the level to the playlist' when the code says `database.AddLevelToPlaylist(level, parentPlaylist);` then it's a sign the comment doesn't need to exist.

```suggestion
            GameLevel? level = database.GetLevelByIdAndType(slotType, slotId);
            if (level == null)
                return NotFound;
 
            database.AddLevelToPlaylist(level, parentPlaylist);
            
            return OK;
```